projects/plugins/protect/src/js/components/fix-all-threats-modal/index.jsx

```diff
@@ -0,0 +1,78 @@
+import { Button, Text } from '@automattic/jetpack-components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { useCallback, useState } from 'react';
+import { STORE_ID } from '../../state/store';
+import ThreatFixHeader from '../threat-fix-header';
+import styles from './styles.module.scss';
+
+const FixAllThreatsModal = threatList => {
+	const { setModal, fixThreats } = useDispatch( STORE_ID );
+	const threatsUpdating = useSelect( select => select( STORE_ID ).getThreatsUpdating() );
+
+	threatList = Object.values( threatList );
```

Comment:
This is something I've been meaning to ask you. We are passing an array to the props in the setModal call, however it gets converted into an object with the array indexes as a key when inside the modal and I'm not really sure why,  that's why I'm doing this conversion here. I couldn't figure out what's modifying the props.

Reply:
This is probably because the modal props get spread into the component, like so:
```javascript
<ModalComponent { ...modalProps } />
```
In this case, modalProps is being set to an array and so the indexes end up as keys. If we make [this adjustment](https://github.com/Automattic/jetpack/pull/26686/files#r995270639), we should be good to go :+1:

I'll add a doc block to the `setModal()` action to better define the modal object properties!

Reply:
Resolved via https://github.com/Automattic/jetpack/pull/26686/commits/f1be1781c70e0a4888f5c57f95bd264d6c7cd8f6